src/cunumeric/index/choose.cc

```diff
@@ -45,6 +45,7 @@ struct ChooseImplBody<VariantKind::CPU, CODE, DIM> {
     } else {
       for (size_t idx = 0; idx < volume; ++idx) {
         auto p = pitches.unflatten(idx, rect.lo);
+        assert(index_arr[p] < choices.size());
```

Comment:
Let's do this only `#ifdef CUNUMERIC_DEBUG`. This property should have already been checked on the python side.